### ISSUE TITLE
Stopped working after several days

### ISSUE THREAD

**Issue by mdhom** [tillsteinbach/WeConnect-mqtt#7]
Hey,

i have the cli running as a docker container container. It worked perfectly, but since two days, no messages are published any more. 
Can i somewhere review the logs what happened?

Thanks a lot,
Max

**Comment by mdhom** [comment]
after restarting container, i see the following logs:
```
weconnect-mqtt_composed | INFO:weconnect-mqtt:Connect to WeConnect
weconnect-mqtt_composed | INFO:weconnect:Token refreshed
weconnect-mqtt_composed | INFO:weconnect-mqtt:Connected to MQTT broker
weconnect-mqtt_composed | INFO:weconnect-mqtt:Update data from WeConnect
weconnect-mqtt_composed | WARNING:weconnect:/vehicles/WVWZZZE1ZMP041013/status/climatisationTimer/timers/1: Unknown attribute singleTimer with value {'startDateTime': '1999-12-31T22:00:00Z'}
weconnect-mqtt_composed | WARNING:weconnect:/vehicles/WVWZZZE1ZMP041013/status/climatisationTimer/timers/2: Unknown attribute singleTimer with value {'startDateTime': '1999-12-31T22:00:00Z'}
weconnect-mqtt_composed | INFO:weconnect-mqtt:New intervall set to 300s by MQTT message
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: /vehicles/WVWZZZE1ZMP041013/status/climatisationSettings/climatisationWithoutExternalPower: new value must be of type <class 'bool'> but is of type <class 'str'>
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: /vehicles/WVWZZZE1ZMP041013/status/climatisationSettings/climatizationAtUnlock: new value must be of type <class 'bool'> but is of type <class 'str'>
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: /vehicles/WVWZZZE1ZMP041013/status/climatisationSettings/windowHeatingEnabled: new value must be of type <class 'bool'> but is of type <class 'str'>
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: /vehicles/WVWZZZE1ZMP041013/status/climatisationSettings/zoneFrontLeftEnabled: new value must be of type <class 'bool'> but is of type <class 'str'>
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: /vehicles/WVWZZZE1ZMP041013/status/climatisationSettings/zoneFrontRightEnabled: new value must be of type <class 'bool'> but is of type <class 'str'>
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: /vehicles/WVWZZZE1ZMP041013/status/climatisationSettings/zoneRearLeftEnabled: new value must be of type <class 'bool'> but is of type <class 'str'>
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: /vehicles/WVWZZZE1ZMP041013/status/climatisationSettings/zoneRearRightEnabled: new value must be of type <class 'bool'> but is of type <class 'str'>
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: id /vehicles/WVWZZZE1ZMP041013/controls/charging cannot be set. You need to provide it in the correct format select one of [start, stop]
weconnect-mqtt_composed | INFO:weconnect-mqtt:Error setting value: id /vehicles/WVWZZZE1ZMP041013/controls/climatization cannot be set. You need to provide it in the correct format select one of [start, stop]
```

**Comment by tillsteinbach** [comment]
Hey, I removed the full VIN from your log (for your privacy). I'll have a look tomorrow. The "Error setting value" should be already fixed in Version 0.6.2
The "singleTimer" I have not seen with my cars before, I will add that to the API as soon as I can, but this should not prevent new messages.

What would be helpful is a log of the container when it does not send anymore.

**Comment by tillsteinbach** [comment]
Can you please try to update to version 0.7.0 and see if the errors are gone?

For the situation where it stopped working I would need a log from that specific situation.

**Comment by mdhom** [comment]
i'll try it, thanks! can i retrieve the logs from some files or must i be attached to the process and view the logs live?

**Comment by tillsteinbach** [comment]
If you use docker-compose you can go with:
```
docker-compose logs weconnect-mqtt
```
if your image has a different name than weconnect-mqtt you have to change that of course.

If not find your container ID with:
```
docker container ls
```
and then execute
```
docker logs <CONTAINER ID>
```

**Comment by tillsteinbach** [comment]
I'll close this for now. If the problem shows again, please re-open

**Comment by mdhom** [comment]
Hi,

the problem reappeared. I now as able to get the logs:
```
weconnect-mqtt_composed | INFO:weconnect:Token refreshed
weconnect-mqtt_composed | Exception in thread Thread-118:
weconnect-mqtt_composed | Traceback (most recent call last):
weconnect-mqtt_composed |   File "/usr/local/lib/python3.9/threading.py", line 954, in _bootstrap_inner
weconnect-mqtt_composed |     self.run()
weconnect-mqtt_composed |   File "/usr/local/lib/python3.9/threading.py", line 1266, in run
weconnect-mqtt_composed |     self.function(*self.args, **self.kwargs)
weconnect-mqtt_composed |   File "/usr/local/lib/python3.9/site-packages/weconnect/weconnect.py", line 374, in __refreshToken
weconnect-mqtt_composed |     raise RetrievalError(f'Status Code from WeConnect server was: {refreshResponse.status_code}')
weconnect-mqtt_composed | weconnect.errors.RetrievalError: Status Code from WeConnect server was: 504
```
I hope that helps?

**Comment by mdhom** [comment]
And there are some other exceptions some time before which do not belong to this issue i think: https://pastebin.com/1kS2rVG7

**Comment by tillsteinbach** [comment]
I'm currently on vacation and will have a look once I'm back!

**Comment by tillsteinbach** [comment]
Should be fixed in 0.12.2. But I cannot reproduce the server errors that lead to the problem. Please reopen if you still experience this crash @mdhom
